PatternPal/PatternPal/Services/StepByStepService.cs

```diff
@@ -1,41 +1,154 @@
-﻿using PatternPal.Core.StepByStep;
-using PatternPal.SyntaxTree;
-using PatternPal.SyntaxTree.Abstractions.Entities;
+﻿#region
+
+using PatternPal.Core.Recognizers;
+using PatternPal.Core.StepByStep;
+
+#endregion
 
 namespace PatternPal.Services;
 
 public class StepByStepService : Protos.StepByStepService.StepByStepServiceBase
 {
+    /// <inheritdoc />
     public override Task< GetInstructionSetsResponse > GetInstructionSets(
         GetInstructionSetsRequest request,
         ServerCallContext context)
     {
-        throw new NotImplementedException();
+        GetInstructionSetsResponse response = new();
+        
+        IDictionary<Recognizer, IRecognizer> sbsDictionary = 
+            RecognizerRunner.SupportedRecognizers;
```

Comment:
This doesn't copy the dictionary, so the `Remove` call on line 32 will modify it!